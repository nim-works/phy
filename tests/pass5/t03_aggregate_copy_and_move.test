discard """
  description: "Make sure the lowering of aggregate assignments works"
"""
(TypeDefs
  (Int 8)
  (Record 8 8
    (Field 0 (Type 0)))
  (ProcTy (Void)))
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 2)
    (List (Type 1) (Type 1))
    (List
      (Block (Params) (Locals) (List)
        (StorageLive (Loc 0) (Loc 1))
        (Asgn (Loc 0) (Copy (Loc 1)))
        (Asgn (Loc 1) (Move (Loc 0)))
        (StorageEnd (Loc 0) (Loc 1))
        (Return)))))
