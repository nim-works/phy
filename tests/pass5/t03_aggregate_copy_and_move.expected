;$sexp
(TypeDefs
  (Int 8)
  (Record 8 8
    (Field 0 (Type 0)))
  (ProcTy (Void))
  (UInt 8))
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 2) 16
    (List
      (Block
        (Params (Type 3))
        (Locals)
        (Store (Type 1)
          (Copy (Param 0))
          (Load (Type 1)
            (Add (Type 3)
              (Copy (Param 0))
              (IntVal 8))))
        (Store (Type 1)
          (Add (Type 3)
            (Copy (Param 0))
            (IntVal 8))
          (Load (Type 1)
            (Copy (Param 0))))
        (Return)))))
